runtime/sema/errors.go

```diff
@@ -3819,3 +3821,11 @@ func (e *ExternalMutationError) SecondaryError() string {
 		e.ContainerType.QualifiedString(),
 	)
 }
+
+type PurityError struct {
```

Comment:
Make sure the error implements the correct interfaces:
```go
var _ SemanticError = &PurityError{}
var _ errors.UserError = &PurityError{}
```

It is a bit strange though, `TestErrorInterfaceConformance` is supposed to catch this and fail 🤔